web/src/app/components/cdf/cdf.component.ts

```diff
@@ -0,0 +1,212 @@
+
+    const durations: number[] = [];
+    pushInfos.forEach(pushInfo => {
+      if (!pushInfo) { return; }
+      const states = pushInfo.stateInfo;
+      if (!states) { return; }
+      const pushEndTime = states[states.length - 1].startTimeNsec;
+      if (!pushEndTime) { return; }
+      const finalState = states[states.length - 1].state;
+      if (!finalState) { return; }
+
+      if (finalState === CdfComponent.COMPLETED_STATE_TAG) {
+        // Find the start time of the first non-empty stage
+        let firstStateStart: number | Long = -1;
+        for (const state of states) {
+          if (state.stage && state.startTimeNsec) {
+            firstStateStart = state.startTimeNsec;
+            break;
+          }
+        }
+        if (firstStateStart !== -1) {
+          const duration = (+pushEndTime - +firstStateStart) / CdfComponent.NANO_TO_MINUTES;
+          durations.push(duration);
+        }
+      }
+    });
+
+    const sortedArray: number[] = durations.sort((n1, n2) => n1 - n2);
+
+    const durationLength = sortedArray.length;
+    for (let i = 0; i < durationLength; i++) {
+      const duration = sortedArray[i];
+      const probability = (i + 1) / durationLength;
+      this.data.push({
+        duration,
+        probability
+      } as CdfData);
+    }
+  }
+
+  /*
+   * Structure of the SVG:
+   * CdfChart:
+   *     - yAxisLeft, yAxisRight, xAxis
+   *     - Axis labels and title
+   *     - Step function CDF plot
+   */
+  private createChart(): void {
+    this.populateData(this.pushInfos);
+
+    const element = this.cdfContainer.nativeElement;
+    const elementWidth = element.clientWidth;
+    const elementHeight = element.clientHeight;
+
+    const margin = { top: 50, right: 50, bottom: 50, left: 50 };
+
+    this.width = elementWidth - margin.left - margin.right;
+    this.height = elementHeight - margin.top - margin.bottom;
+
+    const maxDuration = d3.max(this.data, d => d.duration);
+    if (!maxDuration) { return; }
+
+    this.xScale = d3
+      .scaleLinear()
+      .domain([0, maxDuration + 1])
+      .rangeRound([0, this.width])
+      .nice();
+
+    this.yScale = d3
+      .scaleLinear()
+      .domain([0, 1])
+      .rangeRound([this.height, 0]);
+
+    this.graphData = Array.from(this.data);
+    this.graphData.push({
+      duration: this.xScale.ticks()[this.xScale.ticks().length - 1],
+      probability: 1
+    });
+
+    this.svg = (d3
+      .select(element)
+      .append('svg') as d3SVG)
+      .attr('width', elementWidth)
+      .attr('height', elementHeight);
+
+    const cdfChart = this.svg
+      .append('g')
+      .attr('id', 'chart')
+      .attr('transform', `translate(${margin.left}, ${margin.top})`);
+
+    const yAxisLeft = cdfChart
+      .append('g')
+      .attr('class', 'y-axis-left')
+      .call(d3
+        .axisLeft(this.yScale)
+        .ticks(10)
+        .tickFormat(d3.format(',.1f'))
+      );
+
+    // Removes axis's vertical line and keeps the tick marks
```

Comment:
```suggestion
    // Removes axis's vertical line and keeps the tick marks.
```

Same observation for other places.